From 724b8e7b4f607b510e63fe1bb158b94451effb45 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 16 Jan 2021 14:58:59 -0500
Subject: [PATCH 19/22] sepolicy/treble_app: add rule fix cam2api option

  logs:
      17 00:47:48.796     1     1 W /system/bin/init: type=1107
      audit(0.0:357): uid=0 auid=4294967295 ses=4294967295
      subj=u:r:init:s0 msg='avc: denied { set } for
      property=persist.vendor.camera.eis.enable pid=4411 uid=1000
      gid=1000 scontext=u:r:system_app:s0
      tcontext=u:object_r:vendor_default_prop:s0 tclass=property_service
      permissive=0'

      01-17 00:47:48.796     1     1 W /system/bin/init: type=1107
      audit(0.0:356): uid=0 auid=4294967295 ses=4294967295
      subj=u:r:init:s0 msg='avc: denied { set } for
      property=persist.vendor.camera.HAL3.enabled pid=4411 uid=1000
      gid=1000 scontext=u:r:system_app:s0
      tcontext=u:object_r:vendor_default_prop:s0 tclass=property_service
      permissive=0'
---
 sepolicy/treble_app.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sepolicy/treble_app.te b/sepolicy/treble_app.te
index ef3ea1d..41e6c7d 100644
--- a/sepolicy/treble_app.te
+++ b/sepolicy/treble_app.te
@@ -24,6 +24,8 @@ allow system_app default_hisi_hwservice:hwservice_manager { find };
 type hal_tp_default, domain;
 allow system_app hal_tp_default:binder { call };
 
+#cam2api
+allow system_app vendor_default_prop:property_service { set };
 
 set_prop(system_app, default_prop);
 set_prop(system_app, exported3_default_prop);
-- 
2.17.1

